pandas/_libs/groupby.pyx

```diff
@@ -778,8 +778,13 @@ def group_quantile(ndarray[float64_t] out,
             if not mask[i]:
                 non_na_counts[lab] += 1
 
-    # Get an index of values sorted by labels and then values
-    order = (values, labels)
+    # Get an index of values sorted by labels and then values,
+    # make sure missing labels sort to the back of the array
+    if labels.size:
```

Comment:
Does the subsequent loop even account for missing labels in iteration? If not I wonder if cleaner to just remove rather than force sorting this way

Reply:
Hmm, I think we'd also have to remove from values so that the two have the same shape for lexsort (that might break what's below but I'm not sure honestly)